src/main/java/seedu/weme/model/util/CopyUtil.java

```diff
@@ -0,0 +1,46 @@
+package seedu.weme.model.util;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import seedu.weme.commons.util.FileUtil;
+import seedu.weme.model.imagePath.ImagePath;
+import seedu.weme.model.meme.Meme;
+import seedu.weme.model.template.Template;
+
+/**
+ * A utility class for copying {@code Meme}s and {@code Template}s.
+ */
+public class CopyUtil {
+
+    /**
+     * Copies the image of {@code toCopy} to {@code ImagePath}, and returns a new {@code Meme} whose {@code ImagePath}
+     * points to that location.
+     *
+     * @param toCopy       the {@code Meme} to copy
+     * @param memeLocation the meme image location
+     * @return a new {@code Meme} with the new {@code ImagePath}.
+     */
+    public static Meme copyMeme(Meme toCopy, Path memeLocation) throws IOException {
+        Path originalPath = toCopy.getFilePath().getFilePath();
+        Path newPath = FileUtil.getNewImagePath(originalPath, memeLocation);
+        FileUtil.copy(originalPath, newPath);
```

Comment:
could we abstract this out to another method? Since both meme and template does this